rpc/src/v1/impls/traces.rs

```diff
@@ -18,32 +18,145 @@
 
 use std::sync::{Weak, Arc};
 use jsonrpc_core::*;
-use util::H256;
-use ethcore::client::{BlockChainClient, TransactionID, TraceId};
+use std::collections::BTreeMap;
+use util::{H256, U256, FixedHash, Uint};
+use serde;
+use ethcore::client::{BlockChainClient, CallAnalytics, TransactionID, TraceId};
+use ethcore::trace::VMTrace;
+use ethcore::miner::MinerService;
+use ethcore::state_diff::StateDiff;
+use ethcore::account_diff::{Diff, Existance};
+use ethcore::transaction::{Transaction as EthTransaction, SignedTransaction, Action};
 use v1::traits::Traces;
-use v1::types::{TraceFilter, Trace, BlockNumber, Index};
+use v1::types::{TraceFilter, LocalizedTrace, Trace, BlockNumber, Index, CallRequest, Bytes};
 
 /// Traces api implementation.
-pub struct TracesClient<C> where C: BlockChainClient {
+pub struct TracesClient<C, M> where C: BlockChainClient, M: MinerService {
 	client: Weak<C>,
+	miner: Weak<M>,
 }
 
-impl<C> TracesClient<C> where C: BlockChainClient {
+impl<C, M> TracesClient<C, M> where C: BlockChainClient, M: MinerService {
 	/// Creates new Traces client.
-	pub fn new(client: &Arc<C>) -> Self {
+	pub fn new(client: &Arc<C>, miner: &Arc<M>) -> Self {
 		TracesClient {
 			client: Arc::downgrade(client),
+			miner: Arc::downgrade(miner),
 		}
 	}
+
+	// TODO: share with eth.rs
+	fn sign_call(&self, request: CallRequest) -> Result<SignedTransaction, Error> {
```

Comment:
If you want to share it you can put it to `mod.rs` we have already two functions there.